src/utils/azure_storage.ts

```diff
@@ -161,17 +178,20 @@ export function getBlobAsText(
  * @param blobService     the Azure blob service
  * @param containerName   the name of the Azure blob storage container
  * @param blobName        blob file name
+ * @param defaultValue    if you want a default value insted of BlobNotFound error
  */
 export async function getBlobAsObject<A, O, I>(
   type: t.Type<A, O, I>,
   blobService: azureStorage.BlobService,
   containerName: string,
-  blobName: string
+  blobName: string,
+  defaultValue?: O
```

Comment:
I don't think it makes sense to handle the default here, it should be up to the caller to decide, this function should have the minimal responsibilities - beside you can just call `getOrElse` on the `Either`